pkg/osutil/sharedmem_memfd.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package osutil
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+// CreateMemMappedFile creates a temp file with the requested size and maps it into memory.
+// In the case of Linux, we just use the memfd_create syscall.
+func CreateMemMappedFile(size int) (f *os.File, mem []byte, err error) {
+	// The name is actually irrelevant and can even be the same for all such files.
+	fd, err := unix.MemfdCreate("syz-shared-mem", 0)
```

Comment:
Specialize only how we create the file. The mmap/munmap part is the same for both versions.

Reply:
Moved as much as was possible to `osutil_unix.go`